### ISSUE TITLE
Documentation for fireOnOneCallback parameter of DeferredList is incorrect

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4882]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4882|
|Type|defect|
|Created|2011-02-14 17:14:14Z|
|Branch|https://github.com/twisted/twisted/tree/fireononecallback-4882|

The `__init__` docstring says this about `fireOnOneCallback`:

_When this flag is set, the DeferredList will fire with a two-tuple: the first element is the index in C{deferredList} of the Deferred which fired; the second element is the result of that Deferred._

The implementation of the flag looks like this, though:

```
            if succeeded == SUCCESS and self.fireOnOneCallback:
                self.callback((result, index))
```

<details><summary>Searchable metadata</summary>

```
trac-id__4882 4882
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_fireononecallback_4882 branches/fireononecallback-4882
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1297703654000000 1297703654000000
changetime__1297865721000000 1297865721000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [30757]) Branching to 'fireononecallback-4882'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30758]) Swap the order of the index and result in the documentation

refs [#4882](https://github.com/twisted/twisted/issues/4882)

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to @exarkun|
|-|-|

Awesome! Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [30762]) Merge fireononecallback-4882

Author: exarkun
Reviewer: ralphm
Fixes: [#4882](https://github.com/twisted/twisted/issues/4882)

Correct the API documentation for the `fireOnOneCallback` parameter of `DeferredList`.
